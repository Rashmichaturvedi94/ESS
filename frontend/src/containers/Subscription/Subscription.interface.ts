export interface SubscriptionProps {
  myProp?: boolean;
  /* Add props here. */
}

export interface SubFormValues {
  title: string;
  description: string;
  price: number;
}
