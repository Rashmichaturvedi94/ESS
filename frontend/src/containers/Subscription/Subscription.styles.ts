import styled from 'styled-components';

export const SubContainer = styled.div`
  flex: 1;
  display: flex;
  flex-direction: row;
`;
