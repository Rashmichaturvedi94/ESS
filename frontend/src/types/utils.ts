export type Nullable<T> = {
  [P in keyof T]: T[P] | null;
};

export type Uri = string;

export type DateString = string;

export type LocalOrRemoteFile = File | Uri;

export type Modify<T, R> = Omit<T, keyof R> & R;
