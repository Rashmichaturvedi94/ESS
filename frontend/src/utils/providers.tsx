import React, { ComponentType, FC } from 'react';
import * as Sentry from '@sentry/react';
import { QueryClientProvider } from 'react-query';
import DateFnsUtils from '@date-io/date-fns';
import { ThemeProvider } from 'styled-components';
import { StylesProvider, MuiThemeProvider } from '@material-ui/core';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { queryClient } from '../api';
import { theme } from '../styles/theme';


const createProviderHOC = (Provider: ComponentType) => <P extends object>(
  Component: ComponentType<P>,
) => (props: P) => (
  <Provider>
    <Component {...props} />
  </Provider>
);
const FallbackComponent = () => <div>An error has occurred</div>;

export const Providers: FC = ({ children }) => (
  <Sentry.ErrorBoundary fallback={FallbackComponent} showDialog>
    <StylesProvider injectFirst>
      <MuiThemeProvider theme={theme}>
        <ThemeProvider theme={theme}>
          <MuiPickersUtilsProvider utils={DateFnsUtils}>
            <QueryClientProvider client={queryClient}>
              {children}
            </QueryClientProvider>
          </MuiPickersUtilsProvider>
        </ThemeProvider>
      </MuiThemeProvider>
    </StylesProvider>
  </Sentry.ErrorBoundary>
);

export const withProviders = createProviderHOC(Providers);
