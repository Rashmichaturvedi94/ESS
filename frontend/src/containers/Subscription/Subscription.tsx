import React, { FC } from "react";
import { Box } from "@material-ui/core";
import { useParams } from "react-router-dom";
import { SubContainer } from "./Subscription.styles";
import { useCourse, usePostSubscription } from "../../api";
import { SubscriptionForm } from "../../components/SubscriptionForm";
import { getUserIdFromLocalStorage } from "../../api/api.helpers";
import { Course, CourseParams } from "../../api/api.interface";

export const Subscription: FC = () => {
  const { courseId } = useParams<CourseParams>();
  const course = useCourse({ courseId });
  const { mutate } = usePostSubscription();
  const getImage = (courseObj?: Course) => typeof courseObj?.img === 'string' ? courseObj?.img : '';

  return (
    <SubContainer>
      <Box style={{ marginLeft: 50, marginTop: 50 }}>
        <Box style={{ marginLeft: 40, marginRight: 110 }}>
          <img
            src={getImage(course.data)}
            style={{ height: 450, width: 600, overflow: "hidden" }}
            alt={course.data?.title}
          />
        </Box>
      </Box>

      <SubscriptionForm
        initialValues={{
          title: course.data?.title ?? "tttt",
          description: course.data?.description ?? "tttt",
          price: course.data?.price ?? 10,
          course: course.data?.id ?? 0,
          subscriber: getUserIdFromLocalStorage() ?? 0,
          active: true,
        }}
        onSubmit={(data) => {
          mutate({
            data,
            params: {
              price: course.data?.price,
              course: course.data?.id,
              subscriber: getUserIdFromLocalStorage(),
              active: true,
            },
          });
        }}
      />
    </SubContainer>
  );
};
