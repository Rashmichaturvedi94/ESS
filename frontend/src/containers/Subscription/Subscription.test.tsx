import React from 'react';
import { render } from '../../utils/testUtils';
import { Subscription } from './Subscription';

describe('components/Subscription', () => {
  it('should render', () => {
    const { container } = render(<Subscription />);
    expect(container).toMatchSnapshot();
  });
});
