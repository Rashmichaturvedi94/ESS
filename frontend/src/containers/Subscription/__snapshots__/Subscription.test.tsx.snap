// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Subscription should render 1`] = `
<div>
  <div
    class="sc-bdnxRM bOAqhc"
  >
    <div
      class="MuiBox-root MuiBox-root-1"
      style="margin-left: 50px; margin-top: 50px;"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
        style="margin-left: 40px; margin-right: 110px;"
      >
        <img
          src="https://content.techgig.com/thumb/msid-79844104,width-860,resizemode-4/5-Best-programming-languages-to-learn-in-2021.jpg?140622"
          style="height: 450px; width: 600px; overflow: hidden;"
        />
      </div>
    </div>
    <form>
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <p
          class="MuiTypography-root sc-crzoAE ccmNSu MuiTypography-body1"
        >
          tttt
        </p>
        <label
          class="MuiFormLabel-root MuiInputLabel-root sc-dIsUp RZxpu MuiInputLabel-shrink"
          data-shrink="true"
        >
          tttt
        </label>
        <div
          class="sc-kEqXSa dcUPjU"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            style="font-size: 40px; margin-left: 20px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4v-6h16v6zm0-10H4V6h16v2z"
            />
          </svg>
          <label
            class="MuiFormLabel-root MuiInputLabel-root sc-ksluID lkVrRi MuiInputLabel-shrink"
            data-shrink="true"
          >
            Price:
          </label>
          <label
            class="MuiFormLabel-root MuiInputLabel-root sc-ksluID lkVrRi MuiInputLabel-shrink"
            data-shrink="true"
          >
            $
            10
          </label>
        </div>
        <label
          class="MuiFormLabel-root MuiInputLabel-root sc-bqGGPW caTuyd MuiInputLabel-shrink"
          data-shrink="true"
        >
          Choose Payment method
        </label>
        <div
          class="sc-iqAclL kICOAC"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            style="background: grey; margin-left: 10px;"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Pay by Card
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            style="background: grey; margin-left: 50px;"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Pay by Blik
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
